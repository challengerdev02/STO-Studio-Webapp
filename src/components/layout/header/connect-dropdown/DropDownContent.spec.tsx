import {
  cleanup,
  fireEvent,
  queryAllByRole,
  queryAllByText,
  queryByRole,
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { DropDownContent } from './index';

describe('HybridHeaderComponent', function () {
  afterEach(cleanup);
  const logout = jest.fn();
  let props = {
    //   simulateMobile: false,
    // onFinish : jest.fn(),
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(
      <div {...props} {...extraProps}>
        <DropDownContent logout={logout} />
      </div>
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should display the dd-container', async () => {
    // @ts-ignore
    setup();
    expect(queryByRole(document.body, 'dd-container')).toBeInTheDocument();
  });

  it('should display the header divider', async () => {
    // @ts-ignore
    setup();
    expect(queryAllByRole(document.body, 'separator').length).toBe(3);
  });

  it('should display the header link', async () => {
    // @ts-ignore
    setup();
    expect(queryAllByText(document.body, 'Resource Center').length).toBe(1);
  });

  it('should display the other header links', async () => {
    // @ts-ignore
    setup();
    expect(queryAllByText(document.body, 'My Books').length).toBe(1);
    expect(queryAllByText(document.body, 'My Collections').length).toBe(1);
  });

  it('should logout', async () => {
    // @ts-ignore
    setup();
    const logoutButton = queryAllByText(document.body, 'Log out')[0];
    fireEvent.click(logoutButton);
    expect(logout).toHaveBeenCalled();
  });
});
