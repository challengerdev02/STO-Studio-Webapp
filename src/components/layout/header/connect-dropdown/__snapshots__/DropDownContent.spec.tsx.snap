// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridHeaderComponent should match with snapshot 1`] = `
<div>
  <div>
    <div
      class="container"
      role="dd-container"
    >
      <a
        class="list-item"
        href="/account"
        role="dd-list-item"
      >
        <div>
          Profile
        </div>
      </a>
      <a
        class="list-item"
        href="/account/settings"
        role="dd-list-item"
      >
        <div>
          Profile Settings
        </div>
      </a>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <a
        class="list-item"
        href="/"
        role="dd-list-item"
      >
        <div>
          <i
            class="mc-chart-bar-line mc-lg"
          />
          My Collections
        </div>
      </a>
      <a
        class="list-item"
        href="/"
        role="dd-list-item"
      >
        <div>
          <i
            class="mc-heart-line mc-lg"
          />
          Favorite
        </div>
      </a>
      <a
        class="list-item"
        href="/"
        role="dd-list-item"
      >
        <div>
          <i
            class="mc-design-grid-fill mc-lg"
          />
          My Books
        </div>
      </a>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <a
        class="list-item"
        href="/"
        role="dd-list-item"
      >
        <div>
          <i
            class="mc-bar-01-average-fill mc-lg"
          />
          Resource Center
        </div>
      </a>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <a
        class="list-item"
        href="/account"
        role="dd-list-item"
      >
        <div>
          Account settings
        </div>
      </a>
      <button
        class="ant-btn ant-btn-text list-item"
        role="dd-list-item"
        style="padding-left: 0px; padding-right: 0px;"
        type="button"
      >
        <span>
          Log out
        </span>
      </button>
    </div>
  </div>
</div>
`;
